### DISCUSSION TITLE
Support Images Without a Date

### DISCUSSION THREAD

**Discussion by windmueller** [ente-io/ente#574] [Enhancements]
The web view supports modifying the date for images. Additionally, I would like the option to remove the date completely.

My use-case: I have photos that are several decades old without an accurate timestamp. Displaying an exact date in an album is somewhat confusing. Additionally, mentioning those pictures in "On this day x years ago" is not helpful.

**Comment by vishnukvmd** [comment]
Hey, this is not something that will be easy for us to do right now, since our gallery views are tied to the dates photos were clicked at. What you could do is set an arbitrary time for such photos (like Jan 1st, 1970), and archive the album they are in so they don't appear in memories.

I will however keep this issue open, so that if there are more customers with this use case, they can participate / drive this discussion forward.

**Reply by couvifero** [reply]
Perhaps add a button to set these photos to "unknown date" that just sets them to Jan 1st, 1970 internally?

**Reply by Ronto4** [reply]
I'd fear that this would be very unintuitive for users not having the technical knowledge to know of the significance of 1970-01-01, especially in cases where the known history of photos goes past 1970, so these "unknown date" photos would mix with correctly dated photos.

**Reply by couvifero** [reply]
Fear no more, because obviously it would be trivial for the client to also display an "unknown date" label on these photos instead of 1970-01-01.

**Reply by windmueller** [reply]
@couvifero How would the client handle a photo that has been taken exactly on Thursday, Jan 1st 1970?

**Reply by couvifero** [reply]
At 00:00:00?

What are the odds? You think it's better to have the app deprived of this important functionality for hundreds of real people because in some parallel universe someone might have taken a picture exactly at that time?

Photos didn't even have timestamps at that time, any pictures before 2010 are likely going to be of approximate date and time anyway, which is the feature that Ente is unfortunately missing.

**Reply by windmueller** [reply]
Photos that old do not have exact timestamps or GPS locations. And yes, if someone sets the date manually, leaving the time at midnight might be an option.

Why use an arbitrary date at all and not just an additional "unknown date" flag?

**Reply by couvifero** [reply]
A flag would be better, of course. This thread is just riffing on @vishnukvmd's suggestion of using an arbitrary date.

Even better would be a flag that says a date is "approximate" so we can still specify an year, but not bother with seconds, for example.

**Comment by Ronto4** [comment]
What would be a nice implementation of this feature in my opinion would be to allow incomplete dates. The [Gramps Software](https://gramps-project.org/blog/) allows this for example (notice the lone 1740 without a month or day in this screenshot: https://www.gramps-project.org/wiki/index.php/File:EventsCategory-EventsListView-example-52.png). With this you could specify dates as precisely as possible without resorting to adding dummy dates like the first day of a month.

**Comment by eagle-zero** [comment]
The support of photos without an exact date would be really great and would make ente more useful as a digital photo archive.

I was considering scanning old family photos so that my mother no longer has to search for a specific family album and also has the photos at hand quickly at family gatherings. But since the exact date is often not known, I don't know how to put these photos into ente in a meaningful way. I don't want to label all photos from 1985 with 01.01.1985. It would be nice if you could define a period in which the photo was taken. For example, only the year or year with month or year with season (spring 1985) or decade (80s).

Also a possibility to create a chronological order even without an exact date would be nice. The photos I want to scan are all sorted chronologically in the albums, so I know that a photo on the next page is newer than the one before.

I'm aware that this is most likely not achievable with the current way a date is handled.

**Comment by uhkeller** [comment]
I've just uploaded a bunch of old photos that a relative of mine kindly scanned and sent to me, and I'm encountering the same problem. Having approximate dates would be great.

**Comment by fiatjaf** [comment]
I desperately need something like this. The use case is the same: old photos with not an exact date known. Very often I know the year though, and that is enough.

A solution would be to add (internally) a field on every photo called `approximate`, which then when set would then cause the photos to be displayed as "around 2012".

**Comment by fiatjaf** [comment]
Actually this is also important for recent photos when they are shared by someone else through some other app, for example, and when I receive them they have lost their date-time metadata (for example, they were taken yesterday, but when I get them today they have false metadata from today).

Maybe the `approximate` part should specify a granularity: "hour", "day", "month", "year", because sometimes you know the hour but not the minutes, or you know the day but not the hour -- and so on.

In the edit UI this could be represented as just having a toggle to remove the minutes; remove the timepicker entirely; remove the daypicker and only leave the month, year etc.
